src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/CapiHelper.cs

```diff
@@ -718,8 +718,8 @@ private static void ValidateCspFlags(CspProviderFlags flags)
             if (hr != S_OK)
             {
                 hKey.Dispose();
-                if (IsFlagBitSet((uint)parameters.Flags, (uint)CspProviderFlags.UseExistingKey) ||
-                                                         (uint)hr != (uint)CryptKeyError.NTE_NO_KEY)
+                if (unchecked(IsFlagBitSet((uint)parameters.Flags, (uint)CspProviderFlags.UseExistingKey) ||
```

Comment:
I believe the unchecked statement is only applying on the return value, not the (uint)s, so it should be unchecked{} block instead. Also a tab (vs spaces) was added on the line below.